config.go

```diff
@@ -27,14 +27,28 @@ import (
 	"go.uber.org/zap/zapcore"
 )
 
+// ReportingSamplingConfig offers way to configure reporting of sampling
+// events.
+//
+// With sampling there is always added ambiguity. A record is missing from the
+// logs because it was sampled or because it never took place? Sampling reports
+// intend to compensate for that. See zapcore.NewReportingSampler for details.
+type ReportingSamplingConfig struct {
+	Enabled    bool        `json:"enabled" yaml:"enabled"`
+	LoggerName string      `json:"loggerName" yaml:"loggerName"`
+	Level      AtomicLevel `json:"level" yaml:"level"`
+	Message    string      `json:"message" yaml:"message"`
```

Comment:
I'm not sure whether we want to add a set of fields to represent a log message, or whether we should only provide a simple function hook that returns a log message. otherwise, this may end up expanding significantly to cater to every use-case